README.md

Are you looking for a simple way to display the current network connectivity status in your Flutter app? Check out the "connectivity_plus_widget" package on the Dart programming language's package repository, pub.dev!

This package provides a customizable widget that automatically appears when there is no internet connection and disappears when the connection is restored. You can also customize the properties of the widget and pass your own widget to use instead. Plus, the package allows you to use the connectivity plus widget anywhere in your application.

If you want to simplify the management of network connectivity in your Flutter app, give the "connectivity_plus_widget" package a try. It could save you time and effort in your app development journey!



## Features

* Widget will automatically appears when internet is not connected
* Widget will automatically disappears when internet is connected
* You can customize the properties of connectivity plus widget
* You can pass your own widget
* You can use connectivity plus widget in anywhere in application
* You can also check internet status before calling api or any network operation
* Package also support marquee text 

## Usage

After adding dependency add the below  code in main function to enable connectivity check stream for whole
app

       WidgetsFlutterBinding.ensureInitialized();
       ConnectivityPlusService().initConnectivityService();

## Example
 
    void main() async {
   
       WidgetsFlutterBinding.ensureInitialized();

       ConnectivityPlusService().initConnectivityService();

       runApp(const MyApp());

     }


## Calling Widget
   
    ConnectivityPlusWidget(
            backgroundColor:Colors.red,
            height:40,
            marquee: false,
            textStyle:const TextStyle(fontSize: 20,color: Colors.white),
          ),


    ConnectivityPlusCustomWidget(
            customWidget: const Center( child: Icon(Icons.wifi_off_outlined,color: Colors.red,size: 100,),
           ),
          ),

    
    ConnectivityPlusWidget(
            backgroundColor:Colors.orange,
            height:40,
            marquee: true,

            textStyle:const TextStyle(fontSize: 20,color: Colors.white),
            decoration:  BoxDecoration(
                color: Colors.orange, //new Color.fromRGBO(255, 0, 0, 0.0),
                borderRadius: new BorderRadius.only(
                    topLeft:  const  Radius.circular(20.0),
                    topRight: const  Radius.circular(20.0))
            ),

          ),

## Check Internet Status 


if you want check internet status before calling api or any network operation call the below function ConnectivityPlusService which returns boolean

    ConnectivityPlusService().getInternetStatus()
    log(ConnectivityPlusService().getInternetStatus().toString(),name:"status");


![Sample: ](https://firebasestorage.googleapis.com/v0/b/fluttermania-eebe3.appspot.com/o/connectivity_plus_media%2Fezgif-3-4213079cb3.gif?alt=media&token=2ae8023d-52ae-48e6-825c-94a98ace8bb9)
![Sample: ](https://firebasestorage.googleapis.com/v0/b/fluttermania-eebe3.appspot.com/o/connectivity_plus_media%2F1677491833746.JPEG?alt=media&token=c76ee3bb-10d8-4b2f-ab4d-3278fa17c884)
![Sample: ](https://firebasestorage.googleapis.com/v0/b/fluttermania-eebe3.appspot.com/o/connectivity_plus_media%2F1677493521249.JPEG?alt=media&token=d5fcc1ed-5ebf-40c1-a237-457da3a44112)


## Additional information

for further contact us on flutter.hasan@gmail.com
  



CHANGELOG.md
## 1.0.4

* First release.


lib/connectivity/connectivity_widget.dart
import 'dart:developer';

import 'package:connectivity_plus_widget/connectivity/connectivity_plus_service.dart';
import 'package:flutter/material.dart';
import 'package:get_it/get_it.dart';
import 'package:marquee/marquee.dart';

class ConnectivityPlusWidget extends StatelessWidget {
  final EdgeInsets? padding;
  final Color? backgroundColor;
  final double? height;
  final double? width;
  final TextStyle? textStyle;
  final String? notConnectedText;
  final Decoration? decoration;
  final bool? marquee;

  ConnectivityPlusWidget(
      {Key? key,
      this.backgroundColor,
      this.height,
      this.width,
      this.textStyle,
      this.notConnectedText,
      this.decoration,
      this.padding,
      this.marquee = false})
      : super(key: key);
  final connectivityPlusService = GetIt.I<ConnectivityPlusService>();

  @override
  Widget build(BuildContext context) {
    return StreamBuilder<ConnectivityStatus>(
      stream: connectivityPlusService.connectivityStream,
      initialData: connectivityPlusService.getIntialStatus(),
      builder: (context, snapshot) {
        log("from snapshot", name: snapshot.data.toString());
        if (!snapshot.hasData) {
          return const SizedBox(
            height: 0,
            width: 0,
          );
        } else {
          if (snapshot.data == ConnectivityStatus.wiFi ||
              snapshot.data == ConnectivityStatus.cellular) {
            return const SizedBox(
              height: 0,
              width: 0,
            );
          } else {
            return marquee == false
                ? Container(
                    padding: padding,
                    decoration: decoration,
                    color: decoration == null
                        ? backgroundColor ?? Colors.redAccent
                        : null,
                    height: height ?? 24,
                    width: width ?? double.infinity,
                    child: Center(
                        child: Text(
                      notConnectedText ?? "Internet is not connected",
                      style: textStyle ??
                          const TextStyle(
                              color: Colors.white,
                              fontWeight: FontWeight.w400,
                              fontSize: 16),
                    )))
                : Container(
                    padding: padding,
                    decoration: decoration,
                    color: decoration == null
                        ? backgroundColor ?? Colors.redAccent
                        : null,
                    height: height ?? 24,
                    width: width ?? double.infinity,
                    child: Center(
                        child: Marquee(
                      blankSpace: 25,
                      text: notConnectedText ?? "Internet is not connected",
                      style: textStyle ??
                          const TextStyle(
                              color: Colors.white,
                              fontWeight: FontWeight.w400,
                              fontSize: 16),
                    )));
          }
        }
      },
    );
  }
}

class ConnectivityPlusCustomWidget extends StatelessWidget {
  final Widget customWidget;

  ConnectivityPlusCustomWidget({Key? key, required this.customWidget})
      : super(key: key);
  final connectivityPlusService = GetIt.I<ConnectivityPlusService>();

  @override
  Widget build(BuildContext context) {
    return StreamBuilder<ConnectivityStatus>(
      stream: connectivityPlusService.connectivityStream,
      initialData: connectivityPlusService.getIntialStatus(),
      builder: (context, snapshot) {
        log("from snapshot", name: snapshot.data.toString());
        if (!snapshot.hasData) {
          return const SizedBox(
            height: 0,
            width: 0,
          );
        } else {
          if (snapshot.data == ConnectivityStatus.wiFi ||
              snapshot.data == ConnectivityStatus.cellular) {
            return const SizedBox(
              height: 0,
              width: 0,
            );
          } else {
            return customWidget;
          }
        }
      },
    );
  }
}


lib/connectivity/connectivity_plus_service.dart
import 'dart:async';
import 'dart:developer';
import 'package:connectivity_plus/connectivity_plus.dart';
import 'package:get_it/get_it.dart';

class ConnectivityPlusService {
  bool getInternetStatus() {
    return GetIt.I<ConnectivityPlusService>()._connectedStatus;
  }

  ConnectivityStatus getIntialStatus() {
    if (GetIt.I<ConnectivityPlusService>()._connectedStatus) {
      return ConnectivityStatus.wiFi;
    } else {
      return ConnectivityStatus.offline;
    }
  }

  //late StreamSubscription<ConnectivityResult> _subscription;
  late StreamController<ConnectivityStatus> _streamController;
  late Stream<ConnectivityStatus> connectivityStream;
  ConnectivityStatus _connectivityResult = ConnectivityStatus.init;
  bool _connectedStatus = true;
  void initConnectivityService() {
    GetIt.I
        .registerSingleton<ConnectivityPlusService>(ConnectivityPlusService());
    GetIt.I<ConnectivityPlusService>()._callingInternetStatusCallBackService();
  }

  void _callingInternetStatusCallBackService() async {
    _streamController = StreamController<ConnectivityStatus>.broadcast();

    connectivityStream = _streamController.stream.asBroadcastStream(
      onCancel: (subscription) {
        subscription.pause();
      },
      onListen: (subscription) {
        subscription.resume();
      },
    );
    var checkConnectionForFirstTime =
        await (Connectivity().checkConnectivity());
    _connectivityResult = _getStatusFromResult(checkConnectionForFirstTime);
    _connectedStatus =
        _getConnectedStatusFromResult(checkConnectionForFirstTime);
    log(_connectivityResult.name, name: "logConnectivity");
    _streamController.sink.add(_connectivityResult);

    Connectivity().onConnectivityChanged.listen((ConnectivityResult result) {
      _connectivityResult = _getStatusFromResult(result);
      _connectedStatus = _getConnectedStatusFromResult(result);
      _streamController.sink.add(_connectivityResult);
      log(_connectivityResult.name, name: "logConnectivity");
    });
  }

  ConnectivityStatus _getStatusFromResult(ConnectivityResult result) {
    switch (result) {
      case ConnectivityResult.mobile:
        return ConnectivityStatus.cellular;
      case ConnectivityResult.wifi:
        return ConnectivityStatus.wiFi;
      case ConnectivityResult.none:
        return ConnectivityStatus.offline;
      default:
        return ConnectivityStatus.offline;
    }
  }

  bool _getConnectedStatusFromResult(ConnectivityResult result) {
    switch (result) {
      case ConnectivityResult.mobile:
        return true;
      case ConnectivityResult.wifi:
        return true;
      case ConnectivityResult.none:
        return false;
      default:
        return false;
    }
  }
}

enum ConnectivityStatus { wiFi, cellular, offline, init }


example/main.dart
import 'package:connectivity_plus_widget/connectivity/connectivity_plus_service.dart';
import 'package:connectivity_plus_widget/connectivity/connectivity_widget.dart';
import 'package:flutter/material.dart';
import 'dart:developer';
import 'package:flutter/cupertino.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  ConnectivityPlusService().initConnectivityService();
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Connectivity Plus Widget Example',
      home: Screen1(),
    );
  }
}

class Screen1 extends StatelessWidget {
  const Screen1({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
          body: Column(
        children: [
          ConnectivityPlusWidget(
            backgroundColor: Colors.red,
            height: 40,
            marquee: false,
            textStyle: const TextStyle(fontSize: 20, color: Colors.white),
          ),
          const Spacer(),
          ConnectivityPlusWidget(
            width: 300,
            backgroundColor: Colors.grey,
            textStyle: const TextStyle(fontSize: 20),
          ),
          const Spacer(),
          ConnectivityPlusWidget(
            backgroundColor: Colors.orange,
            height: 100,
            notConnectedText: "Custom Text",
            marquee: true,
            textStyle: const TextStyle(fontSize: 30, color: Colors.white),
            decoration: BoxDecoration(
              color: Colors.blueAccent,
              border: Border.all(
                width: 10,
                color: Colors.red,
              ),
            ),
          ),
          const Spacer(),
          ConnectivityPlusCustomWidget(
            customWidget: Center(
              child: Container(
                child: const Center(child: Text("Custom Widget ")),
                height: 100,
                width: 300,
                decoration: BoxDecoration(
                  color: Colors.greenAccent,
                  border: Border.all(
                    width: 10,
                    color: Colors.black,
                  ),
                ),
              ),
            ),
          ),
          const Spacer(),
          CupertinoButton(
              child: const Text("Click here to check internet status"),
              onPressed: () {
                log(ConnectivityPlusService().getInternetStatus().toString(),
                    name: "status");
              }),
          Container(
            margin: const EdgeInsets.all(20),
            child: ConnectivityPlusWidget(
              notConnectedText: "No Internet",
              marquee: false,
              padding: const EdgeInsets.all(10),
              backgroundColor: Colors.orange,
              height: 50,
              textStyle: const TextStyle(fontSize: 20, color: Colors.red),
              decoration: new BoxDecoration(
                color: Colors.amberAccent, //new Color.fromRGBO(255, 0, 0, 0.0),
                borderRadius: new BorderRadius.all(
                  const Radius.circular(40.0),
                ),
              ),
            ),
          ),
          const Spacer(),
          ConnectivityPlusCustomWidget(
            customWidget: const Center(
              child: Icon(
                Icons.wifi_off_outlined,
                color: Colors.red,
                size: 100,
              ),
            ),
          ),
          const Spacer(),
          ConnectivityPlusWidget(
            backgroundColor: Colors.orange,
            height: 40,
            marquee: true,
            textStyle: const TextStyle(fontSize: 20, color: Colors.white),
            decoration: BoxDecoration(
                color: Colors.orange, //new Color.fromRGBO(255, 0, 0, 0.0),
                borderRadius: new BorderRadius.only(
                    topLeft: const Radius.circular(20.0),
                    topRight: const Radius.circular(20.0))),
          ),
          ConnectivityPlusWidget(
            backgroundColor: Colors.red,
            height: 40,
            marquee: true,
            textStyle: const TextStyle(fontSize: 20, color: Colors.white),
          ),
        ],
      )),
    );
  }
}
